webrender/src/device/gfx/descriptor.rs

```diff
@@ -377,10 +381,10 @@ where
         range: std::ops::Range<usize>,
         sampler_linear: &B::Sampler,
         sampler_nearest: &B::Sampler,
-    ) {
-        let new_set = match self.descriptor_bindings.entry(bindings) {
-            Entry::Occupied(_) => return,
-            Entry::Vacant(v) => {
+    ) -> DescriptorSet<B> {
+        let new_set = match self.descriptor_bindings.remove(&bindings) {
```

Comment:
Perhaps in followup PRs, we need to avoid hashmap lookups done so often. Conceptually, what you have is indeed a map, but you know beforehand all the keys, and it's not that many (just the primitive/clip/other, right?). In this case, we could just implement this as a `match` over the key, and the compiler would be able to avoid any memory allocation and generate much better code.